iface-ng/iface-ng

```diff
@@ -0,0 +1,146 @@
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#------------------------------------------------------------------------
+
+usage() {
+    echo "Usage: $0 [-d down] [-n \"no ip\"] [-c FF0000] [-C 00FF00] [-e] [-p] [-s] [-6]
+    -d down: Placeholder text if the specified interface is down. If set to an empty string,
+       the entire block is hidden (useful to save space on small screens).
+    -n \"no ip\": Placeholder text to display when the interface is up but it does not
+       have any ip address.
+    -c FF0000: Text color when interface is down.
+    -C 00FF00: Text color when interface is up.
+    -e: Show interface even if it does not exist (useful for tun/tap).
+    -p: Include CIDR prefix.
+    -s: Include link speed (ethernet only).
+    -6: Use ipv6 address insead of v4." 1>&2
+    exit 1; 
+}
+
+# Default options
+downText="down"
+noipText="no ip"
+colorDown="FF0000"
+colorUp="00FF00"
+inet="inet"
+e=0
+p=0
+s=0
+
+while getopts "d:n:c:C:esp6" opt; do
+    case "${opt}" in
+        d)
+            downText=${OPTARG}
+            ;;
+        n)
+            noipText=${OPTARG}
+            ;;
+        c)
+            colorDown=${OPTARG#\#}
+            ;;
+        C)
+            colorUp=${OPTARG#\#}
+            ;;
+        e)
+            e=1
+            ;;
+        p)
+            p=1
+            ;;
+        s)
+            s=1
+            ;;
+        6)
+            inet="inet6"
+            ;;
+        *)
+            usage
+            ;;
+        esac
+done
+
+# Use the provided interface, otherwise the device used for the default route.
+if [[ -n $BLOCK_INSTANCE ]]; then
+    IF=$BLOCK_INSTANCE
+else
+    IF=$(ip route | awk '/^default/ { print $5 ; exit }')
+fi
+
+#------------------------------------------------------------------------
+
+if [[ ! -d /sys/class/net/${IF} ]]; then
+    if [[ $e -eq 1 ]]; then
+        echo $downText # full text
+        echo $downText # short text
+        echo "#$colorDown" # color
+    fi
+
+    exit
+fi
+
+
+#------------------------------------------------------------------------
+
+if [[ "$(cat /sys/class/net/$IF/operstate)" = 'down' ]]; then
+    if [[ $downText = '' ]]; then
+        exit
+    fi
+    echo $downText # full text
+    echo $downText # short text
+    echo "#$colorDown" # color
+    exit
+fi
+
+regex=" +inet6? ("
+
+if [[ $inet = "inet6" ]]; then
+    regex="$regex([a-fA-F0-9]{0,4}:?){2,8}"
+else
+    regex="$regex([0-9]+\.?){4}"
+fi
+
+if [[ $p -eq 1 ]]; then
+    regex="$regex(\/[0-9]+)?"
+else
+    regex="$regex"
+fi
+
+regex="$regex).*"
+
+IPADDR=$(ip addr show dev $IF | grep "$inet " | sed -r "s/$regex/\1/")
+
+
+if [[ $IPADDR = '' ]]; then
+    IPADDR=$noipText
+fi
+
+if [[ $s -eq 1 ]]; then
+    speed=" ($(cat /sys/class/net/$IF/speed) Mbps)"
+else
+    speed=""
+fi
+
+case $BLOCK_BUTTON in
```

Comment:
Sorry, I don't see what line are exactly referring to, but the `IPADDR=$noipText` is documented (`-n` option), and wether to show link speed is also documented (`-s` option).